File: cmd_gift.py
# ملف: cmd_gift.py
from datetime import datetime, timedelta
from db_manager import get_user, update_user_gold
from aiogram import types

COMMANDS = ["هدية"]
DAY = timedelta(hours=24)  # مدة الانتظار 24 ساعة

# =========================
# التحقق إذا المستخدم يقدر ياخذ هديته
# =========================
async def can_take_gift(uid):
    user = await get_user(uid)
    last_gift_time = user.get("last_gift_time", None)
    if not last_gift_time:
        return True
    last_gift = datetime.fromtimestamp(last_gift_time)
    return datetime.now() - last_gift >= DAY

# =========================
# استلام الهدية
# =========================
async def take_gift(uid, amount=100):
    if not await can_take_gift(uid):
        return None
    await update_user_gold(uid, amount)
    # تحديث وقت آخر هدية
    user = await get_user(uid)
    user["last_gift_time"] = datetime.now().timestamp()
    # هنا لو تستخدم قاعدة بيانات فعلية احفظ الوقت فيها
    return await get_user(uid)["gold"]

# =========================
# التعامل مع الأمر
# =========================
async def handle(message: types.Message, bot):
    text = message.text.strip()
    if text not in COMMANDS:
        return

    uid = message.from_user.id

    if await can_take_gift(uid):
        amount = await take_gift(uid)
        await bot.send_message(
            message.chat.id,
            f"🎁 لقد استلمت هديتك اليومية!\n+{amount} ذهب 💰\n✨ مفاجأة اليوم لك!"
        )
        return

    # =========================
    # حساب الوقت المتبقي
    # =========================
    user = await get_user(uid)
    last_gift_time = user.get("last_gift_time", 0)
    last_gift = datetime.fromtimestamp(last_gift_time)
    remaining = DAY - (datetime.now() - last_gift)
    if remaining.total_seconds() < 0:
        remaining = timedelta(seconds=0)

    hours = remaining.seconds // 3600
    minutes = (remaining.seconds % 3600) // 60
    await bot.send_message(
        message.chat.id,
        f"⏳ انتظر قبل استلام هديتك القادمة:\n {hours} ساعة و {minutes} دقيقة"
    )

File: cmd_gift.js
// ملف: cmd_gift.js
const COMMANDS = ["هدية"];
const DAY = 24 * 60 * 60 * 1000; // 24 ساعة بالميلي ثانية
const db_manager = require("./db_manager"); // تأكد من مسار الملف

// =========================
// التحقق إذا المستخدم يقدر ياخذ هديته
// =========================
async function canTakeGift(uid) {
    const user = await db_manager.getUser(uid);
    const lastGift = user.last_gift_time || 0;
    return Date.now() - lastGift >= DAY;
}

// =========================
// استلام الهدية
// =========================
async function takeGift(uid, amount = 100) {
    if (!(await canTakeGift(uid))) return null;

    await db_manager.updateUserGold(uid, amount);
    await db_manager.users.updateOne({ uid }, { $set: { last_gift_time: Date.now() } });
    return await db_manager.getUserGold(uid);
}

// =========================
// التعامل مع الأمر
// =========================
async function handle(bot, msg) {
    if (!msg.text || !COMMANDS.includes(msg.text.trim())) return;

    const uid = msg.from.id;

    if (await canTakeGift(uid)) {
        const amount = await takeGift(uid);
        bot.sendMessage(msg.chat.id, `🎁 لقد استلمت هديتك اليومية!\n+${amount} ذهب 💰`);
        return;
    }

    // =========================
    // حساب الوقت المتبقي
    // =========================
    const user = await db_manager.getUser(uid);
    const lastGift = user.last_gift_time || 0;
    let remaining = DAY - (Date.now() - lastGift);
    if (remaining < 0) remaining = 0;

    const hours = Math.floor(remaining / (1000 * 60 * 60));
    const minutes = Math.floor((remaining % (1000 * 60 * 60)) / (1000 * 60));
    // const seconds = Math.floor((remaining % (1000 * 60)) / 1000); // لو حبيت تضيف الثواني

    bot.sendMessage(
        msg.chat.id,
        `⏳ انتظر قبل استلام هديتك القادمة:\n ${hours} ساعة و ${minutes} دقيقة`
        // + ` و ${seconds} ثانية` // لو تريد الثواني
    );
}

module.exports = { handle, canTakeGift, takeGift };
